odoo/addons/base/models/res_users.py

```diff
@@ -662,11 +669,7 @@ def context_get(self):
         }
 
         # ensure the language is set and is compatible with the web client
-        lang = context.get('lang') or (request and request.default_lang()) or DEFAULT_LANG
-        if lang == 'ar_AR':
-            context['lang'] = 'ar'
-        if lang in babel.core.LOCALE_ALIASES:
-            context['lang'] = babel.core.LOCALE_ALIASES[lang]
+        context['lang'] = get_lang(self.env, context.get('lang')).code
```

Comment:
You can use `_get_cached('code`)` (on the lang record) instead to use a ormcache. The extra query should come from here.

Reply:
"The extra query should come from here" -> I was wrong obviously. 

`get_lang` need more data than the previous version:
- It used two ORM cached methods: `_lang_get` (cache with `_lang_get_id`) and `get_installed`
- But it also need to access to `user.company_id.partner_id.lang` which can trigger 3 SQL request (if the land is not present in the context) ...